blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -1001,6 +1002,50 @@ fn sync_tipset<DB: BlockStore + Sync + Send + 'static, C: Consensus>(
     })
 }
 
+async fn fetch_batch<DB: BlockStore + Send + Sync + 'static, C: Consensus>(
+    batch: &[Arc<Tipset>],
+    network: &SyncNetworkContext<DB>,
+    chainstore: &ChainStore<DB>,
+    s: &channel::Sender<FullTipset>,
+) -> Result<(), TipsetRangeSyncerError<C>> {
+    // Tipsets in `batch` are already in chronological order
+    let head = batch.last().unwrap(); // This is safe because a batch can't be empty
```

Comment:
We can avoid the panic entirely.

```suggestion
    if let head = batch.last() {
```